TypeCobol/Compiler/Diagnostics/CodeElementCheckers.cs

```diff
@@ -346,10 +346,10 @@ class SetStatementForIndexesChecker
     {
         public static void OnCodeElement(SetStatementForIndexes set, CodeElementsParser.SetStatementForIndexesContext context)
         {
-            if (set.SendingVariable == null)
+            var sendingVariable = set.SendingVariable;
+            if (sendingVariable == null)
             {
-                DiagnosticUtils.AddError(set, "Set xxx up/down by xxx: Sending field missing or type unknown",
-                    context?.variableOrExpression2());
+                DiagnosticUtils.AddError(set, "Set xxx up/down by xxx: Sending field missing or type unknown", context);
```

Comment:
Is it intended to have a part of the checks here and an other part in `TypeCobol/Compiler/Diagnostics/TypeCobolChecker.cs`?

Reply:
The other part of the check needs SymbolResoution and is specific to TypeCobol language, that's why it's in `TypeCobolChecker.cs`.

Reply:
Ok, that makes sense then.